tests/system-tests/rdscore/internal/rdscorecommon/sriov-pod-level-bond.go

```diff
@@ -0,0 +1,1114 @@
+
+	podDeployment = podDeployment.WithNodeSelector(nodeSelector)
+
+	podDeployment = podDeployment.WithSecondaryNetwork(netAnnotations)
+
+	return podDeployment
+}
+
+func generateTCPTraffic(clientPod *pod.Builder, serverIPAddr, serverPort string) (string, error) {
+	By("Generating TCP traffic")
+
+	glog.V(100).Infof("Ensure pod %q in namespace %q is Ready",
+		clientPod.Definition.Name, clientPod.Definition.Namespace)
+
+	err := clientPod.WaitUntilReady(5 * time.Second)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to wait for pod %q in namespace %q to become Ready: %v",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+
+		return "", fmt.Errorf("failed to wait for pod %q in namespace %q to become Ready: %w",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+	}
+
+	cmdToRun := []string{"bash", "-c",
+		fmt.Sprintf("testcmd -protocol tcp -port %s -interface net3 -server %s",
+			serverPort, serverIPAddr)}
+
+	glog.V(100).Infof("Execute command: %q", cmdToRun)
+
+	var output string
+
+	err = wait.PollUntilContextTimeout(
+		context.TODO(),
+		time.Second*5,
+		time.Minute*1,
+		true,
+		func(ctx context.Context) (bool, error) {
+			result, err := clientPod.ExecCommand(cmdToRun, clientPod.Object.Spec.Containers[0].Name)
+
+			if err != nil {
+				glog.V(100).Infof("Error running command from within a pod %q: %v",
+					clientPod.Object.Name, err)
+
+				return false, nil
+			}
+
+			glog.V(100).Infof("Successfully executed command from within a pod %q in namespace %q",
+				clientPod.Object.Name, clientPod.Definition.Namespace)
+
+			output = result.String()
+			glog.V(100).Infof("Command's output:\n\t%v", output)
+
+			return true, nil
+		})
+
+	if err != nil {
+		return "", fmt.Errorf("failed to run command from within pod %s: %w", clientPod.Object.Name, err)
+	}
+
+	return output, nil
+}
+
+func generateLongTCPTraffic(
+	clientPod *pod.Builder,
+	serverIPAddr,
+	serverPort,
+	packetsNumber,
+	timeout string) (string, error) {
+	By("Open long TCP session and generate traffic")
+
+	glog.V(100).Infof("Ensure pod %q in namespace %q is Ready",
+		clientPod.Definition.Name, clientPod.Definition.Namespace)
+
+	err := clientPod.WaitUntilReady(5 * time.Second)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to wait for pod %q in namespace %q to become Ready: %v",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+
+		return "", fmt.Errorf("failed to wait for pod %q in namespace %q to become Ready: %w",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+	}
+
+	cmdToRun := []string{"bash", "-c",
+		fmt.Sprintf("testcmd -protocol tcp -port %s -interface net3 -packages %s -timeoutTCP %s -server %s",
+			serverPort, packetsNumber, timeout, serverIPAddr)}
+
+	glog.V(100).Infof("Execute command: %q", cmdToRun)
+
+	var output string
+
+	err = wait.PollUntilContextTimeout(
+		context.TODO(),
+		time.Second*3,
+		time.Minute,
```

Comment:
how does this work with the `-packages` and `-timeoutTCP` params from the _cmdToRun_ ?

For e.g. if one specifies `-timeoutTCP 10min`, will this _wait.PollUntilContextTimeout_ work?

Reply:
-timeoutTCP is not a 10 min, but 10 sec.
tcp timeout meaning is a **connection** TCP timeout (how long to retransmit and leave TCP session open when no response from the other side), recommendations are different, the testcmd app default is 2 sec.